spec/sections/09-core.adoc

```diff
@@ -459,7 +461,13 @@ geo:hasGeometry
     a rdf:Property, owl:ObjectProperty ;
     rdfs:isDefinedBy geo: ;
     rdfs:domain geo:Feature ;
-    rdfs:range geo:Geometry ;    
+    rdfs:range
+		[
+			owl:unionOf (
+				:Geometry
+				:GeometryCollection
+			)
+		] ;
```

Comment:
I think using a union for rdfs:range here takes us out of OWL2 RL profile. We probably need a larger discussion for this one.

Reply:
Are you sure that `range -> UnionClassExpression` isn't allowed in RL?

My reading of the [OWL RL Class Expressions](https://www.w3.org/TR/2012/REC-owl2-profiles-20121211/#Class_Expressions_3) grammar is that class union expressions_are_ allowed.

The grammar line for RL's object property allowances states

```
subObjectUnionOf := 'ObjectUnionOf' '(' subClassExpression subClassExpression { subClassExpression } ')'
```

indicates that an RL-legal "object union of" can be declared for (arbitrary) subclasses expressions, with subclass expressions themselves being

```
subClassExpression :=
    Class other than owl:Thing |
    subObjectIntersectionOf | subObjectUnionOf | ObjectOneOf |
    subObjectSomeValuesFrom | ObjectHasValue |
    DataSomeValuesFrom | DataHasValue
```

which means union on `ClassX ClassY` would be fine.

Reply:
Looks like UnionClassExpressions are allowed in general, but the spec mentions this restriction.

> OWL 2 RL axioms about property expressions are as in the structural specification [[OWL 2 Specification](https://www.w3.org/TR/2012/REC-owl2-profiles-20121211/#ref-owl-2-specification)], the only differences being that class expressions in property domain and range axioms are restricted to superClassExpression, and that the use of reflexive properties is disallowed.

`ObjectPropertyRange := 'ObjectPropertyRange' '(' axiomAnnotations ObjectPropertyExpression superClassExpression ')'`

From what I can tell, superClassExpression does not include these union expressions.
```
superClassExpression :=
    Class other than owl:Thing |
    superObjectIntersectionOf | superObjectComplementOf |
    superObjectAllValuesFrom | ObjectHasValue | superObjectMaxCardinality |
    DataAllValuesFrom | DataHasValue | superDataMaxCardinality
```
It seems like we may need a concrete superclass of Geometry and GeometryCollection, which feels kind of awkward.

Reply:
If we need such classes my suggestion would be: geo:{SpatialObject|Feature|Geometry}Manifestation whereas geo:SpatialObjectManifestation is the superclass of the other two and geo:Geometry and geo:GeometryCollection are subclasses of geo:GeometryManifestation.

But I agree with Matt. Would be good if we could avoid this class hierarchy.